auth-foundation/src/main/java/com/okta/authfoundation/client/internal/NetworkUtils.kt

```diff
@@ -127,36 +127,40 @@ private suspend fun OidcConfiguration.executeRequest(
 ): Response {
     currentCoroutineContext().ensureActive()
     val rateLimitErrorCode = 429
+    var event: RateLimitExceededEvent
+    var retryCount = 0
+    var response: Response
+    var requestToExecute = request
+    var delaySeconds = 0L
+
+    do {
+        delay(delaySeconds.seconds)
+        response = okHttpClient.newCall(requestToExecute).await()
+
+        if (response.code != rateLimitErrorCode) return response
+
+        val timeToResetSeconds: Long
+        val requestId = response.header("X-Okta-Request-Id")
+        val responseTimeSeconds = response.headers.getDate("Date")?.time?.milliseconds?.inWholeSeconds
+        val retryTimeSeconds = response.header("X-Rate-Limit-Reset")?.toLongOrNull()
+        if (responseTimeSeconds == null || retryTimeSeconds == null) {
+            return response
+        }
+        response.use {
```

Comment:
Do we need this `response.use`?

Reply:
There could be a resource leak otherwise. A more explicit way to write what I want is to do ```response.close()``` in a try catch block, if that's preferred

Reply:
I'm now using a ```response.close()```